app/assets/v2/js/lib/casper/casper_js_sdk.js

```diff
@@ -0,0 +1 @@
t t=this.options.curve.keyFromPrivate(e,"hex").getPublic("hex");i=this.privateKeyObject(e,t)}else if("der"===r){if("string"!=typeof e);else{if("string"!=typeof e)throw"private key must be a buffer or a string";e=t.from(e,"hex")}i=a.decode(e,"der")}else{if("pem"!==r)throw"invalid private key format";if("string"!=typeof e)throw"private key must be a string";i=a.decode(e,"pem",this.options.privatePEMOptions)}if("raw"===n)return i.privateKey.toString("hex");if("der"===n)return a.encode(i,"der").toString("hex");if("pem"===n)return a.encode(i,"pem",this.options.privatePEMOptions);throw"invalid destination format for private key"}encodePublic(e,r,n){let i;if("raw"===r){if("string"!=typeof e)throw"public key must be a string";i=this.publicKeyObject(e)}else if("der"===r){if("string"!=typeof e);else{if("string"!=typeof e)throw"public key must be a buffer or a string";e=t.from(e,"hex")}i=s.decode(e,"der")}else{if("pem"!==r)throw"invalid public key format";if("string"!=typeof e)throw"public key must be a string";i=s.decode(e,"pem",this.options.publicPEMOptions)}if("raw"===n)return i.pub.data.toString("hex");if("der"===n)return s.encode(i,"der").toString("hex");if("pem"===n)return s.encode(i,"pem",this.options.publicPEMOptions);throw"invalid destination format for public key"}}r.default=f,f.ECPrivateKeyASN=a,f.SubjectPublicKeyInfoASN=s}).call(this)}).call(this,e("buffer").Buffer)},{"asn1.js":26,"bn.js":45,buffer:266,elliptic:56}],102:[function(e,t,r){"use strict";var n=e("inherits"),i=e("hash-base"),o=e("safe-buffer").Buffer,a=new Array(16);function s(){i.call(this,64),this._a=1732584193,this._b=4023233417,this._c=2562383102,this._d=271733878}function u(e,t){return e<<t|e>>>32-t}function f(e,t,r,n,i,o,a){return u(e+(t&r|~t&n)+i+o|0,a)+t|0}function c(e,t,r,n,i,o,a){return u(e+(t&n|r&~n)+i+o|0,a)+t|0}function h(e,t,r,n,i,o,a){return u(e+(t^r^n)+i+o|0,a)+t|0}function l(e,t,r,n,i,o,a){return u(e+(r^(t|~n))+i+o|0,a)+t|0}n(s,i),s.prototype._update=function(){for(var e=a,t=0;t<16;++t)e[t]=this._block.readInt32LE(4*t);var r=this._a,n=this._b,i=this._c,o=this._d;r=f(r,n,i,o,e[0],3614090360,7),o=f(o,r,n,i,e[1],3905402710,12),i=f(i,o,r,n,e[2],606105819,17),n=f(n,i,o,r,e[3],3250441966,22),r=f(r,n,i,o,e[4],4118548399,7),o=f(o,r,n,i,e[5],1200080426,12),i=f(i,o,r,n,e[6],2821735955,17),n=f(n,i,o,r,e[7],4249261313,22),r=f(r,n,i,o,e[8],1770035416,7),o=f(o,r,n,i,e[9],2336552879,12),i=f(i,o,r,n,e[10],4294925233,17),n=f(n,i,o,r,e[11],2304563134,22),r=f(r,n,i,o,e[12],1804603682,7),o=f(o,r,n,i,e[13],4254626195,12),i=f(i,o,r,n,e[14],2792965006,17),r=c(r,n=f(n,i,o,r,e[15],1236535329,22),i,o,e[1],4129170786,5),o=c(o,r,n,i,e[6],3225465664,9),i=c(i,o,r,n,e[11],643717713,14),n=c(n,i,o,r,e[0],3921069994,20),r=c(r,n,i,o,e[5],3593408605,5),o=c(o,r,n,i,e[10],38016083,9),i=c(i,o,r,n,e[15],3634488961,14),n=c(n,i,o,r,e[4],3889429448,20),r=c(r,n,i,o,e[9],568446438,5),o=c(o,r,n,i,e[14],3275163606,9),i=c(i,o,r,n,e[3],4107603335,14),n=c(n,i,o,r,e[8],1163531501,20),r=c(r,n,i,o,e[13],2850285829,5),o=
```

Comment:
we might need to move this to the node_module !
@gdixon could you throw in some documentation on how we can go about doing this ?

Reply:
we exempted cross-chain libs from npm 'cause some of the files have been modified by us. e.g polkadot and tezos